apps/aecore/src/aec_paying_for_tx.erl

```diff
@@ -122,9 +122,15 @@ check(#paying_for_tx{}, Trees, _Env) ->
     %% Checks in process/3
     {ok, Trees}.
 
--spec signers(tx(), aec_trees:trees()) -> {ok, [aec_keys:pubkey()]}.
-signers(#paying_for_tx{} = Tx, _) ->
-    {ok, [payer_pubkey(Tx)]}.
+-spec signers(tx(), aec_trees:trees()) ->
+    {ok, [aec_keys:pubkey()]} | {error, inner_tx_incorrectly_signed}.
+signers(#paying_for_tx{tx = STx} = Tx, Trees) ->
+    case aetx_sign:verify(STx, Trees, ?IRIS_PROTOCOL_VSN, <<"-inner_tx">>) of
```

Comment:
I am a bit concerned with the hardcoded protocol version here.

Reply:
I added a soothing comment :-)